src/scanner.c

```diff
@@ -74,6 +74,8 @@
 
 #define PEEK env->lexer->lookahead
 
+#define ARR_SIZE(x) ((sizeof(x) / sizeof(0[x])) / ((size_t)(!(sizeof(x) % sizeof(0[x])))))
```

Comment:
What's the purpose of this? Are you ensuring that the array is nonempty?

What's the deal with writing `0[x]` instead of `x[0]`?

Reply:
This is the best version I found on the internet, taken from https://stackoverflow.com/questions/4415524/common-array-length-macro-for-c.

I'd usually define macro as `sizeof(x) / sizeof(x[0])` but advantages of writing it as in PR are:

```
It improves on the array[0] or *array version by using 0[array], which is equivalent to array[0] on plain arrays, but will fail to compile if array happens to be a C++ type that overloads operator[]().

The division causes a divide-by-zero operation (that should be caught at compile time since it's a compile-time constant expression) for many (but not all) situations where a pointer is passed as the array parameter.
```

Reply:
Given that we only use it for this one array, and that we know that it's non-empty, I think we could just hardcode something for readability.

It surely is more resilient though, so if you want to keep it, at least write a comment explaining what it does exactly!